website/docs/developer/research/research-deep-glacier-migration.md

```diff
@@ -0,0 +1,226 @@
+
+  rule {
+    id = "transition to deep archive"
+    status = "Enabled"
+
+    transition {
+      days          = 0
+      storage_class = "DEEP_ARCHIVE"
+    }
+    filter {
+      prefix = "tmp/"
+    }
+  }
+
+```
+:::tip
+To migrate to a different storage class if needed, change `storage_class` under transition block to the desired class in the terraform code above.
+:::
+
+### Migrating specific ORCA holdings to deep archive.
+
+S3 lifecycle rule will not work in this case. 
+Migration should be a two step process- first retrieve the objects and then copy the objects to deep archive.
+
+1. User inputs a list of granuleId/collectionId or a combination of both via Cumulus console/API call.
+2. This API should trigger a lambda function, Fargate or even S3 batch operations that queries the ORCA catalog and retrives the files via `Bulk Retrieval` and copies the objects to the glacier bucket as versioned object. In addition, it should also create and update the migration job status in the ORCA catalog. The technology used should depend upon the number of bulk requests made. If amount of bulk requests is large, then lambda function might throw timeout errors. In this case, S3 batch operations or Fargate could be a better alternative. Additional research is needed on this. 
+<br />
+Bulk retrievals are typically completed within 5–12 hours. The prices are lowest among all retrieval rates - $0.025 per 1,000 requests.
+3. Once files are restored in the bucket, the bucket event notification for `ObjectRestore:Completed` action will trigger a lambda function that will copy the restored object to deep archive storage. In addition, this lambda should also update the ORCA catalog with `Deep Archive` storage type value. Note that the current existing restore workflow trigger should be disabled during the whole migration process to prevent triggering the `request_files` lambda and then re-enabled it migration is complete.
+Another option here is to update existing recovery lambdas to perform the above tasks instead of creating new lambda. Additional research is needed to choose the right option.
+
+The following is an initial workflow and architecture for ORCA deep archive migration for specific files. Note that this could possibly change during implementation phase.
+
+<MyImage
+    imageSource={useBaseUrl('img/Deep-Archive-Migration-workflow-specific-files.svg')}
+    imageAlt="System Context"
+    zoomInPic={useBaseUrl('img/zoom-in.png')}
+    zoomOutPic={useBaseUrl('img/zoom-out.png')}
+    resetPic={useBaseUrl('img/zoom-pan-reset.png')}
+/>
+
+<br />
+<br />
+<br />
+
+<MyImage
+    imageSource={useBaseUrl('img/ORCA-Deep-Archive-Migration-Architecture-specific-files.svg')}
+    imageAlt="System Context"
+    zoomInPic={useBaseUrl('img/zoom-in.png')}
+    zoomOutPic={useBaseUrl('img/zoom-out.png')}
+    resetPic={useBaseUrl('img/zoom-pan-reset.png')}
+/>
+
+<br />
+<br />
+<br />
+
+One option of bulk retrieval is to use python boto3 client for S3:
```

Comment:
Move line 134 - 166 under step 2 (line 102).

Update sentence to something like 
```
An example of initiating AWS bulk retrieval of objects using Python and the boto3 client for S3 can be seen below.
```